api/objects.proto

```diff
@@ -257,6 +257,15 @@ message NetworkAttachment {
 	repeated string aliases = 3;
 }
 
+message CNMState {
+	// Driver specific operational state provided by the network driver.
+	Driver driver_state = 4;
+
+	// Runtime state of IPAM options. This may not reflect the
+	// ipam options from NetworkSpec.
+	IPAMOptions ipam = 5 [(gogoproto.customname) = "IPAM"];
```

Comment:
Now that these are moving into their own message, the field numbers can start from 1.